### ISSUE TITLE
{Linked:Bug13061061; Linked:Bug13061064}[Adaptive Card- Designer] - Screen Reader is not announcing the changed state information while navigating on the 'Delta' button.

### ISSUE THREAD

**Issue by vagpt** [microsoft/AdaptiveCards#9152]
### Target Platforms

Other

### SDK Version

1.6

### Application Name

Adaptive Cards

### Problem Description

**Test Environment:**
OS Build: Windows 11
Version: 24H2(OS Build25276.3281)  
Browser: Edge-dev
URL: https://adaptivecards.io/designer/
Browser Version: Edge Dev 136.0.3193.0

**Pre-Reqisite:**
1. Paste the Json on the 'Card payload Editor' section.

**Repro Steps:**
1. Open the above URL on the EDGE Dev.
2. Press tab key and navigate Preview button and press enter key to activate it.
3. Press tab key and navigate to the 'Delta' button and observe the issue.

**Actual Result:**
While navigating on the 'Delta 20.00 USD' button, screen reader is not announcing the state information whether the button is collapsed or expand.

Also, on activating it using enter key screen reader is not announcing the changed information as well.

**Expected Result:**
While navigating on the 'Delta 20.00 USD' button, screen reader should announce the state information as well. It should announce the state as collapsed or expanded when focus lands on it.

**Note:**
Same issue is repro with NVDA.

**User Impact:**
It will impact the screen reader user as they will not able to know whether the control is expand state or collapse state also after activating it also screen reader user is not able to know whether the action is taken or not?

**WCAG Reference:**
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships

Have feedback to share on Bugs ? please tag bug as “A11yRCA” and add your feedback in the comment section

### Screenshots

https://github.com/user-attachments/assets/21f98db2-1204-4534-bf69-a1e902bf1d9a

![Image](https://github.com/user-attachments/assets/dd8036a7-33d9-4b21-9baa-518dc3615123)

### Card JSON

```JSON
{
	"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
	"body": [
		{
			"items": [
				{
					"actions": null,
					"columns": [
						{
							"items": [
								{
									"size": "Medium",
									"spacing": null,
									"style": "heading",
									"text": "Displaying(1 of 1) Expenses",
									"type": "TextBlock",
									"weight": "Bolder",
									"wrap": true
								},
								{
									"size": null,
									"spacing": "None",
									"style": null,
									"text": "Teams limits message size. Displaying up to 7 recent expenses. Expenses with ❌ are missing required receipts. Verify expense details to submit.",
									"type": "TextBlock",
									"weight": "Lighter",
									"wrap": true
								}
							],
							"type": "Column",
							"width": 2
						}
					],
					"inlines": null,
					"items": null,
					"separator": null,
					"type": "ColumnSet"
				}
			],
			"separator": null,
			"spacing": null,
			"type": "Container"
		},
		{
			"items": [
				{
					"actions": null,
					"columns": null,
					"inlines": null,
					"items": [
						{
							"columns": [
								{
									"horizontalAlignment": null,
									"items": [],
									"selectAction": null,
									"spacing": null,
									"type": "Column",
									"verticalContentAlignment": null,
									"width": 2
								}
							],
							"id": null,
							"isVisible": null,
							"items": null,
							"type": "ColumnSet",
							"verticalContentAlignment": null,
							"width": null
						},
						{
							"columns": [
								{
									"horizontalAlignment": null,
									"items": [
										{
											"altText": null,
											"ariaLabel": null,
											"horizontalAlignment": null,
											"id": null,
											"isVisible": null,
											"size": "Default",
											"text": "DELTA | 20.00 USD | 12/17/2024",
											"type": "TextBlock",
											"url": null,
											"weight": "Bolder",
											"width": null,
											"wrap": true
										}
									],
									"selectAction": null,
									"spacing": null,
									"type": "Column",
									"verticalContentAlignment": null,
									"width": 84
								},
								{
									"horizontalAlignment": null,
									"items": [
										{
											"altText": null,
											"ariaLabel": null,
											"horizontalAlignment": "Right",
											"id": null,
											"isVisible": null,
											"size": null,
											"text": "✅",
											"type": "TextBlock",
											"url": null,
											"weight": null,
											"width": null,
											"wrap": true
										}
									],
									"selectAction": null,
									"spacing": null,
									"type": "Column",
									"verticalContentAlignment": null,
									"width": 8
								},
								{
									"horizontalAlignment": "Right",
									"items": [
										{
											"altText": "click to expand",
											"ariaLabel": "click to expand",
											"horizontalAlignment": "Right",
											"id": "5662716825_Down",
											"isVisible": null,
											"size": null,
											"text": null,
											"type": "Image",
											"url": "https://adaptivecards.io/content/down.png",
											"weight": null,
											"width": "20px",
											"wrap": null
										},
										{
											"altText": "click to collapse",
											"ariaLabel": "click to collapse",
											"horizontalAlignment": "Right",
											"id": "5662716825_Up",
											"isVisible": false,
											"size": null,
											"text": null,
											"type": "Image",
											"url": "https://adaptivecards.io/content/up.png",
											"weight": null,
											"width": "20px",
											"wrap": null
										}
									],
									"selectAction": {
										"horizontalAlignment": "Right",
										"targetElements": [
											"5662716825_Content",
											"5662716825_Down",
											"5662716825_Up"
										],
										"title": "DELTA | 20.00 USD | 12/17/2024",
										"type": "Action.ToggleVisibility"
									},
									"spacing": "small",
									"type": "Column",
									"verticalContentAlignment": "center",
									"width": 8
								}
							],
							"id": null,
							"isVisible": null,
							"items": null,
							"type": "ColumnSet",
							"verticalContentAlignment": "center",
							"width": "stretch"
						},
						{
							"columns": null,
							"id": "5662716825_Content",
							"isVisible": false,
							"items": [
								{
									"columns": [
										{
											"horizontalAlignment": null,
											"items": [
												{
													"actions": null,
													"spacing": null,
													"text": "Category",
													"type": "TextBlock",
													"weight": "Bolder",
													"wrap": true
												},
												{
													"actions": null,
													"spacing": "None",
													"text": "Airfare",
													"type": "TextBlock",
													"weight": null,
													"wrap": true
												}
											],
											"type": "Column",
											"width": 60
										},
										{
											"horizontalAlignment": null,
											"items": [
												{
													"actions": null,
													"spacing": null,
													"text": "Payment Type",
													"type": "TextBlock",
													"weight": "Bolder",
													"wrap": true
												},
												{
													"actions": null,
													"spacing": "None",
													"text": "Cash",
													"type": "TextBlock",
													"weight": null,
													"wrap": true
												}
											],
											"type": "Column",
											"width": 40
										}
									],
									"type": "ColumnSet"
								},
								{
									"columns": [
										{
											"horizontalAlignment": null,
											"items": [
												{
													"actions": null,
													"spacing": null,
													"text": "Additional Info",
													"type": "TextBlock",
													"weight": "Bolder",
													"wrap": true
												},
												{
													"actions": null,
													"spacing": "None",
													"text": "test",
													"type": "TextBlock",
													"weight": null,
													"wrap": true
												}
											],
											"type": "Column",
											"width": 60
										},
										{
											"horizontalAlignment": null,
											"items": [
												{
													"actions": null,
													"spacing": null,
													"text": "Receipt(s)",
													"type": "TextBlock",
													"weight": "Bolder",
													"wrap": true
												},
												{
													"actions": null,
													"spacing": "None",
													"text": "",
													"type": "TextBlock",
													"weight": null,
													"wrap": true
												}
											],
											"type": "Column",
											"width": 40
										}
									],
									"type": "ColumnSet"
								},
								{
									"columns": [
										{
											"horizontalAlignment": "Center",
											"items": [
												{
													"actions": [
														{
															"associatedInputs": "auto",
															"data": {
																"actionSubmitId": "Cash",
																"value": "{\"merchantName\": \"Delta\", \"transactionAmount\": 20,\n \"isReceiptsAttached\": false,\n \"isReceiptRequired\": false,\n \"paymentType\": \"Cash\",\n \"lineItemDescription\": \"test\",\n \"transactionCurrency\": \"USD\", \"transactionDate\": \"2024-12-17T12:00:00\", \"transactionId\": \"\",\n \"lineItemId\": \"5662716825\", \"expenseCategory\": \"Airfare\"\n }"
															},
															"title": "Edit",
															"tooltip": null,
															"type": "Action.Submit"
														}
													],
													"spacing": null,
													"text": null,
													"type": "ActionSet",
													"weight": null,
													"wrap": null
												}
											],
											"type": "Column",
											"width": 30
										},
										{
											"horizontalAlignment": "Center",
											"items": [
												{
													"actions": [
														{
															"associatedInputs": "auto",
															"data": {
																"actionSubmitId": "Remove",
																"value": "{\"merchantName\": \"Delta\", \"transactionAmount\":\"20\", \"paymentType\":\"Cash\", \"transactionDate\": \"2024-12-17T12:00:00\", \"transactionId\":\"\", \"lineItemId\": \"5662716825\", \"expenseCategory\": \"Airfare\"}"
															},
															"title": "Exclude",
															"tooltip": "Remove from expense report",
															"type": "Action.Submit"
														}
													],
													"spacing": null,
													"text": null,
													"type": "ActionSet",
													"weight": null,
													"wrap": null
												}
											],
											"type": "Column",
											"width": 35
										},
										{
											"horizontalAlignment": "Center",
											"items": [
												{
													"actions": [
														{
															"associatedInputs": "auto",
															"data": {
																"actionSubmitId": "Add Receipt",
																"value": "{\"merchantName\": \"Delta\", \"transactionAmount\":\"20\", \"paymentType\":\"Cash\", \"transactionDate\": \"2024-12-17T12:00:00\", \"transactionId\":\"\", \"lineItemId\": \"5662716825\", \"expenseCategory\": \"Airfare\"}"
															},
															"title": "Add Receipt",
															"tooltip": null,
															"type": "Action.Submit"
														}
													],
													"spacing": null,
													"text": null,
													"type": "ActionSet",
													"weight": null,
													"wrap": null
												}
											],
											"type": "Column",
											"width": 35
										}
									],
									"type": "ColumnSet"
								},
								{
									"columns": [
										{
											"horizontalAlignment": null,
											"items": [],
											"type": "Column",
											"width": 2
										}
									],
									"type": "ColumnSet"
								}
							],
							"type": "Container",
							"verticalContentAlignment": null,
							"width": null
						}
					],
					"separator": true,
					"type": "Container"
				}
			],
			"separator": true,
			"spacing": null,
			"type": "Container"
		},
		{
			"items": [
				{
					"actions": [
						{
							"associatedInputs": "auto",
							"data": null,
							"spacing": "Medium",
							"title": "Create An Expense Report",
							"type": "Action.Submit",
							"wrap": null
						}
					],
					"columns": null,
					"inlines": null,
					"items": null,
					"separator": null,
					"type": "ActionSet"
				}
			],
			"separator": true,
			"spacing": "Medium",
			"type": "Container"
		},
		{
			"items": [
				{
					"actions": [
						{
							"associatedInputs": "auto",
							"data": {
								"actionSubmitId": "Complex Expenses",
								"value": "Complex Expenses"
							},
							"spacing": null,
							"title": "(0) Complex Expenses",
							"type": "Action.Submit",
							"wrap": true
						}
					],
					"columns": null,
					"inlines": null,
					"items": null,
					"separator": null,
					"type": "ActionSet"
				},
				{
					"actions": null,
					"columns": null,
					"inlines": [
						{
							"color": null,
							"selectAction": null,
							"text": "Complex expenses require additional actions and include Hotel Itemization, Guest Names, Per Diems, etc. and are not currently supported. Complete reconciliation directly in ",
							"type": "TextRun",
							"weight": "Lighter"
						},
						{
							"color": "Accent",
							"selectAction": {
								"title": "Click the link to access MyExpense and manage your expenses",
								"type": "Action.OpenUrl",
								"url": "https://myexpense.operations.dynamics.com/"
							},
							"text": "MyExpense",
							"type": "TextRun",
							"weight": "Lighter"
						},
						{
							"color": null,
							"selectAction": null,
							"text": ".",
							"type": "TextRun",
							"weight": "Default"
						}
					],
					"items": null,
					"separator": null,
					"type": "RichTextBlock"
				}
			],
			"separator": true,
			"spacing": "Medium",
			"type": "Container"
		}
	],
	"type": "AdaptiveCard",
	"version": "1.5"
}
```

### Sample Code

Originally tracking bug id - 

https://dev.azure.com/microsoftit/OneITVSO/_workitems/edit/13061061

**Comment by vagpt** [comment]
Tracking bug id - 

https://dev.azure.com/microsoftit/OneITVSO/_workitems/edit/13061064

**Comment by vagpt** [comment]
Closing this issue as dependent team bug is closed.
